code/__HELPERS/cmp.dm

```diff
@@ -0,0 +1,38 @@
+/proc/cmp_numeric_dsc(a,b)
+	return b - a
+
+/proc/cmp_numeric_asc(a,b)
+	return a - b
+
+/proc/cmp_text_asc(a,b)
+	return sorttext(b,a)
+
+/proc/cmp_text_dsc(a,b)
+	return sorttext(a,b)
+
+/proc/cmp_name_asc(atom/a, atom/b)
+	return sorttext(b.name, a.name)
+
+/proc/cmp_name_dsc(atom/a, atom/b)
+	return sorttext(a.name, b.name)
+
+/proc/cmp_ckey_asc(client/a, client/b)
+	return sorttext(b.ckey, a.ckey)
+
+/proc/cmp_ckey_dsc(client/a, client/b)
+	return sorttext(a.ckey, b.ckey)
+
+/proc/cmp_subsystem_init(datum/controller/subsystem/a, datum/controller/subsystem/b)
+	return initial(b.init_order) - initial(a.init_order)	//uses initial() so it can be used on types
+
+/proc/cmp_subsystem_display(datum/controller/subsystem/a, datum/controller/subsystem/b)
+	return sorttext(b.name, a.name)
+
+/proc/cmp_subsystem_priority(datum/controller/subsystem/a, datum/controller/subsystem/b)
+	return a.priority - b.priority
+
+/proc/cmp_atom_layer_asc(atom/A,atom/B)
+	if(A.plane != B.plane)
+		return A.plane - B.plane
+	else
+		return A.layer - B.layer
```

Comment:
seems to be missing a whole bunch of entries, but I dunno how useful they are.

Reply:
Anything removed in here didn't compile because they referenced non-existent typepaths

Reply:
alrighty.